db/pebble/transaction.go

```diff
@@ -123,17 +123,13 @@ func (t *Transaction) Impl() any {
 // NewIterator : see db.Transaction.NewIterator
 func (t *Transaction) NewIterator() (db.Iterator, error) {
 	var iter *pebble.Iterator
-	var err error
 	if t.batch != nil {
-		iter, err = t.batch.NewIter(nil)
+		iter = t.batch.NewIter(nil)
 	} else if t.snapshot != nil {
-		iter, err = t.snapshot.NewIter(nil)
+		iter = t.snapshot.NewIter(nil)
```

Comment:
what happened with prev error ? 🤔 (i.e. do we need to call some new method to check for it)

Reply:
The error was added after we manually upgraded on its `main` branch. Apparently that behavior wasn't backported to the `crl-release-23.1` branch.